src/portable/portable.c

```diff
@@ -1425,21 +1425,32 @@ static bool marker_matches_images(const char *marker, const char *name_or_path,
                         if(!(e[strspn(e, "/")] == 0 || streq(e, ".raw")))
                                 return false;
                 } else {
-                        const char *b, *underscore;
+                        const char *b, *e, *underscore;
                         size_t l;
 
                         /* We shall match against a path. Let's ignore any prefix here though, as often there are many ways to
-                        * reach the same file. However, in this mode, let's validate any file suffix. */
+                         * reach the same file. However, in this mode, let's validate any file suffix. But also ensure that we
+                         * don't fail if both components don't have a '/' at all (strcspn returns the full length of the string
+                         * in that case, which might not match as the versions might differ). */
 
+                        e = strchr(a, '/');
                         l = strcspn(a, "/");
                         b = last_path_component(*image_name_or_path);
 
-                        if (strcspn(b, "/") != l)
+                        if (!!e != !!strchr(b, '/')) /* One is a directory, the other is not */
```

Comment:
why not use `(a[l] != '/') == !strchr(b, '/'))` as test? if so, you don't need e at all..

Reply:
done - but with `!=` (if one has the `/` and the other hasn't, then its' not a match and we return false), and `e` is necessary as the next check needs to be skipped if it's not defined